src/python/pants/backend/python/dependency_inference/rules.py

```diff
@@ -245,6 +247,12 @@ def _get_inferred_asset_deps(
 
         if inferred_tgts:
             possible_addresses = tuple(tgt.address for tgt in inferred_tgts)
+            if len(possible_addresses) == 1:
+                resolve_results[filepath] = ImportResolveResult(
+                    ImportOwnerStatus.unambiguous, possible_addresses
+                )
+                continue
```

Comment:
I feel like this cluster of if-elses might be clearer if every branch sets a local variable `ImportOwnerStatus`, and then has a single line to update `resolve_results` with that value. Otherwise the type checker isn't actually helping to guarantee that all branches mutate the dict.

Reply:
agree. I've refactored to use a function and comprehension.